lib/api/root_api.rb

```diff
@@ -328,6 +334,7 @@ def self.authentication_scope(sym)
       authenticate
       set_localization
       enforce_content_type
+      skip_session_write
```

Comment:
This should be first, as authenticate can raise and session will still be written
```suggestion
      skip_session_write
      authenticate
      set_localization
      enforce_content_type
```
Seems to at least fix the flaky test at spec/features/users/create_spec.rb:120